project-example/src/main.rs

```diff
@@ -15,10 +15,15 @@ use std::collections::BTreeMap;
 use structopt::StructOpt;
 use tap::*;
 
-mod generated_code;
-mod lib;
+mod connection_handler;
+mod opt;
 
-use generated_code::example;
+// HACK: Include `generated.rs` directly rather than treating it like a regular
+// module file because it internally declares a module named `generated`, which
```

Comment:
Btw, the internal module can be named whatever we want. Everything gets re-exported at the bottom anyway. It's just there to be a "root" module for all the inner modules to refer back to. We could potentially remove the nested module and use the file itself as the root module, not sure.

Reply:
I think it makes sense to remove that root module, since it makes the structure of the generated file non-idiomatic, and gives the consuming code less control over the module structure. I don't want to do that as part of this PR, so I'll leave it as-is and it can be cleaned up later.